Mathlib/Topology/Order/ScottTopology.lean

```diff
@@ -0,0 +1,439 @@
+/-
+Copyright (c) 2023 Christopher Hoskin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christopher Hoskin
+-/
+
+import Mathlib.Logic.Equiv.Defs
+import Mathlib.Order.Directed
+import Mathlib.Order.UpperLower.Basic
+import Mathlib.Topology.Basic
+import Mathlib.Topology.ContinuousFunction.Basic
+import Mathlib.Topology.Order
+import Mathlib.Topology.Order.Lattice
+import Mathlib.Topology.Order.UpperLowerSetTopology
+import Mathlib.Topology.Order.LowerUpperTopology
+
+/-!
+# Scott topology
+
+This file introduces the Scott topology on a preorder.
+
+## Main definitions
+
+- `inaccessible_by_directed_joins` - a set `u` is said to be inaccessible by directed joins if, when
+  the least upper bound of a directed set `d` lies in `u` then `d` has non-empty intersection with
+  `u`.
+- `ScottTopology'` - the Scott topology is defined as the join of the topology of upper sets and the
+  Scott-Hausdorff topology (the topological space where a set `u` is open if, when the least upper
+  bound of a directed set `d` lies in `u` then there is a tail of `d` which is a subset of `u`).
+
+## Main statements
+
+- `ScottTopology.isOpen_isUpper` - Scott open sets are upper.
+- `ScottTopology.isClosed_isLower` - Scott closed sets are lower.
+- `ScottTopology.continuous_monotone` - functions continuous wrt the Scott topology are
+  monotone.
+- `ScottTopology.scottContinuous_iff_continuous` - a function is Scott continuous (preserves least
+  upper bounds of directed sets) if and only if it is continuous wrt the Scott topology.
+- `ScottTopology.instT0Space` - the Scott topology on a partial order is T₀.
+
+## Implementation notes
+
+A type synonym `WithScottTopology` is introduced and for a preorder `α`, `WithScottTopology α`
+is made an instance of `topological_space` by the `ScottTopology'`.
+
+We define a mixin class `ScottTopology` for the class of types which are both a preorder and a
+topology and where the topology is the `ScottTopology'`.
+It is shown that `WithScottTopology α` is an instance of `ScottTopology`.
+
+A class `Scott` is defined in `topology.omega_complete_partial_order` and made an instance of a
+topological space by defining the open sets to be those which have characteristic functions which
+are monotone and preserve limits of countable chains. Whilst this definition of the Scott topology
+coincides with the one given here in some special cases, in general they are not the same
+[Domain Theory, 2.2.4][abramsky_gabbay_maibaum_1994].
+
+## References
+
+* [Gierz et al, *A Compendium of Continuous Lattices*][GierzEtAl1980]
+* [Abramsky and Jung, *Domain Theory*][abramsky_gabbay_maibaum_1994]
+
+## Tags
+Scott topology, preorder
+-/
+
+variable (α β : Type _)
+
+open Set
+
+section preorder
+
+variable {α} {β}
+
+variable [Preorder α] [Preorder β]
+
+def inaccessible_by_directed_joins (u : Set α) : Prop :=
```

Comment:
This is not following the new naming convention.

Reply:
Is that documented somewhere? The link from the README appears to still be for lean3 mathlib https://leanprover-community.github.io/contribute/naming.html

Reply:
I'm assuming it should be `InaccessibleByDirectedJoins`?